back/app/models/idea.rb

```diff
@@ -171,6 +171,31 @@ def custom_form
     end
   end
 
+  def input_term
+    return project.input_term if project.continuous?
+
+    return creation_phase.input_term if participation_method_on_creation.form_in_phase?
+
+    participation_context = ParticipationContextService.new.get_participation_context(project)
```

Comment:
I wonder if we shouldn't separate all the code that follows here. For example, to return the input term of a project when there is no current phase. Do you think it could make sense to introduce an `InputTermService`?

Reply:
I wouldn't do that, because this method is only relevant for ideas. Extracting the code would not add value, because the code cannot reused for other objects. Only for ideas.